src/main/java/com/Elessar/app/server/LogOffHandler.java

```diff
@@ -36,19 +41,22 @@ public void handle(HttpExchange he) throws IOException {
             final LogoffRequest logoffRequest = LogoffRequest.parseFrom(is);
             final LogoffResponse.Builder logoffResponse = LogoffResponse.newBuilder();
             final String userName = logoffRequest.getName();
-            if (!userData.containsKey(userName)) {
-                logoffResponse.setSuccess(false).setFailReason(userName + " is NOT a Registered User !");
+            User prevUser = db.update(new User(userName, null, null, null, "false"));
```

Comment:
final

And don't we need to try-catch the database update?

Reply:
findoneandupdate doesn't seem throwing an exception, the document says it will return null if cannot find. What do we need to catch then?

Reply:
alright